doc/library/tensor/nnet/conv.txt

```diff
@@ -102,68 +87,37 @@ TODO: Give examples on how to use these things! They are pretty complicated.
-      CPU, nnet.conv2d will be replaced by CorrMM. To explicitly disable it, set
-      ``THEANO_FLAGS=optimizer_excluding=conv_gemm`` in your environment.
+      `caffe's cpp implementation <https://github.com/BVLC/caffe/blob/master/src/caffe/layers/conv_layer.cpp>`_.
+      It does not flip the kernel.
     - :func:`dnn_conv <theano.gpuarray.dnn.dnn_conv>` GPU-only
-      convolution using NVIDIA's cuDNN library. This requires that you have
-      cuDNN 4.0 or newer installed and available, which in turn requires CUDA 7.0
-      and a GPU with compute capability 3.0 or more.
-
-      If cuDNN is available, by default, Theano will replace all nnet.conv2d
-      operations with dnn_conv. To explicitly disable it, set
-      ``THEANO_FLAGS=optimizer_excluding=conv_dnn`` in your environment.
-      As dnn_conv has a gradient defined, you can also use it manually.
+      convolution using NVIDIA's cuDNN library.
+
 - Implemented operators for neural network 3D / video convolution:
     - :func:`conv3D <theano.tensor.nnet.Conv3D.conv3D>`
-      3D Convolution applying multi-channel 3D filters to batches of
-      multi-channel 3D images. It does not flip the kernel.
+      Old convolution that is removed.
     - :func:`GpuCorr3dMM <theano.gpuarray.blas.GpuCorr3dMM>`
       This is a GPU-only 3d correlation relying on a Toeplitz matrix
       and gemm implementation (see :func:`GpuCorrMM <theano.sandbox.cuda.blas.GpuCorrMM>`)
       It needs extra memory for the Toeplitz matrix, which is a 2D matrix of shape
       ``(no of channels * filter width * filter height * filter depth, output width * output height * output depth)``.
-      As it provides a gradient, you can use it as a replacement for nnet.conv3d.
-      Alternatively, you can use nnet.conv3d and allow Theano's graph optimizer
-      to replace it by the GEMM version by setting
-      ``THEANO_FLAGS=optimizer_including=conv3d_gemm:convgrad3d_gemm:convtransp3d_gemm`` in your environment.
-      This is not enabled by default because it uses some extra memory, but the
-      overhead is small compared to conv3d_fft, there are no restrictions on
-      input or kernel shapes and strides are supported. If using it,
-      please see the warning about a bug in CUDA 5.0 to 6.0
-      in :func:`GpuCorrMM <theano.sandbox.cuda.blas.GpuCorrMM>`.
-
     - :func:`Corr3dMM <theano.tensor.nnet.corr3d.Corr3dMM>`
       This is a CPU-only 3d correlation implementation based on
       the 2d version (:func:`CorrMM <theano.tensor.nnet.corr.CorrMM>`).
       It does not flip the kernel. As it provides a gradient, you can use it as a
       replacement for nnet.conv3d. For convolutions done on CPU,
-      nnet.conv3d will be replaced by Corr3dMM. To explicitly disable it, set
-      ``THEANO_FLAGS=optimizer_excluding=conv_gemm`` in your environment.
+      nnet.conv3d will be replaced by Corr3dMM.
 
     - :func:`dnn_conv <theano.gpuarray.dnn.dnn_conv>` GPU-only
```

Comment:
At last suggestion: I think we should reference dnn_conv3d here (as we are in conv3d section):
```
    - :func:`dnn_conv3d <theano.gpuarray.dnn.dnn_conv3d>` GPU-only
```